### PR TITLE
Enable Swift Package Index documentation and add API doc link

### PR DESCRIPTION
**Pull request by Kyle-Ye**
Bug/issue #, if applicable: None

## Summary

1. Add .spi.yml file to make SwiftPackageIndex build documentation for this package.
2. Update README.md to add a link pointing to the corresponding link. (The link will be available on the next day after the PR is merged. SPI's CI is triggered with a 24h window.)

> See https://github.com/apple/swift-algorithms/blob/main/.spi.yml

## Dependencies

None

## Testing

None

## Checklist

- [x] Ran the `swift test` and it succeeded
- [x] Updated documentation if necessary

## Future work

After this PR, we'll only get the basic docc documentation support.

- [ ] Add `SystemPackage.docc` in `Sources/System` to adjust the doc page.

### CODE REVIEW

**Comment by Kyle-Ye** [comment]
@swift-ci please test

File: .spi.yml
Comment (MaxDesiatov) [inline .spi.yml:4]: nit: missing newline
```suggestion
    - documentation_targets: [SystemPackage]

```

**Comment by Kyle-Ye** [comment]
@swift-ci please test

**Comment by MaxDesiatov** [comment]
@Kyle-Ye would you kindly update the description providing more context and motivation for the change? Thanks!

**Comment by Kyle-Ye** [comment]
> @Kyle-Ye would you kindly update the description providing more context and motivation for the change? Thanks!

Done. Do you think we should address the "Future work" part in this PR?

**Comment by MaxDesiatov** [comment]
IMO no need to get the PR bigger than it is now, let's wait for feedback from maintainers first.

**Review by glessard — APPROVED** [review]

**Pull request merged**
